Errors when trying NPM publish

When running `npm publish --access public` getting two errors:

1) Missing ABI folder --> Fixed by running `hh export-abi` in the prepublish script
2) `esModuleInterop` error below

```
tomi@ceres:/U/t/Doc/g/th/contracts (dev) ✗ npm publish --access public

> @graphprotocol/contracts@1.14.0-testnet-product prepublishOnly
> scripts/prepublish

yarn run v1.22.19
$ rm -rf build/ cache/ dist/
✨  Done in 0.07s.
yarn run v1.22.19
$ SKIP_LOAD=true scripts/build
$ hardhat compile
...

Generating typings for: 94 artifacts in dir: build/types for target: ethers-v5
Successfully generated 151 typings!
Compiled 91 Solidity files successfully
✨  Done in 14.00s.
Nothing to compile
No need to generate any newer typings.
node_modules/hardhat/node_modules/ethereumjs-util/dist/externals.d.ts:6:8 - error TS1259: Module '"/Users/tomi/Documents
/git/thegraph/contracts/node_modules/@types/bn.js/index"' can only be default-imported using the 'esModuleInterop' flag

6 import BN from 'bn.js';
         ~~

  node_modules/@types/bn.js/index.d.ts:594:1
    594 export = BN;
        ~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleIn
terop' flag.

node_modules/rlp/dist/types.d.ts:2:8 - error TS1259: Module '"/Users/tomi/Documents/git/thegraph/contracts/node_modules/
@types/bn.js/index"' can only be default-imported using the 'esModuleInterop' flag

2 import BN from 'bn.js';
         ~~

  node_modules/@types/bn.js/index.d.ts:594:1
    594 export = BN;
        ~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleIn
terop' flag.


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/hardhat/node_modules/ethereumjs-util/dist/externals.d.ts:6
     1  node_modules/rlp/dist/types.d.ts:2
npm ERR! code 2
npm ERR! path /Users/tomi/Documents/git/thegraph/contracts
npm ERR! command failed
npm ERR! command sh -c scripts/prepublish

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tomi/.npm/_logs/2022-07-25T09_24_14_879Z-debug-0.log
```

--- a/hardhat.config.ts
+++ b/hardhat.config.ts
@@ -155,8 +155,9 @@ const config: HardhatUserConfig = {
   },
   abiExporter: {
     path: './build/abis',
-    clear: false,
+    clear: true,
     flat: true,
+    runOnCompile: true,
   },
   tenderly: {
     project: 'graph-network',

--- a/scripts/prepublish
+++ b/scripts/prepublish
@@ -7,6 +7,7 @@ set -eo pipefail
 # Build contracts
 yarn clean
 yarn build
+npx hardhat export-abi
 
 # Refresh distribution folder
 rm -rf dist && mkdir -p dist/types/_src
@@ -16,7 +17,7 @@ cp -R build/types/ dist/types/_src
 ### Build Typechain bindings
 
 # Build and create TS declarations
-tsc -d ${TYPECHAIN_DIR}/_src/*.ts --outdir ${TYPECHAIN_DIR}
+tsc -d ${TYPECHAIN_DIR}/_src/*.ts --outdir ${TYPECHAIN_DIR} --esModuleInterop
 # Copy back sources
 cp ${TYPECHAIN_DIR}/_src/*.ts ${TYPECHAIN_DIR}
 # Delete temporary src dir

--- a/scripts/prepublish
+++ b/scripts/prepublish
@@ -7,7 +7,6 @@ set -eo pipefail
 # Build contracts
 yarn clean
 yarn build
-npx hardhat export-abi
 
 # Refresh distribution folder
 rm -rf dist && mkdir -p dist/types/_src

